sqlmesh/dbt/adapter.py

```diff
@@ -219,7 +219,14 @@ def list_relations(self, database: t.Optional[str], schema: str) -> t.List[BaseR
         reference_relation = self.relation_type.create(
             database=database, schema=schema, quote_policy=self.quote_policy
         )
-        return self.list_relations_without_caching(reference_relation)
+        normalized_schema = self._normalize(self._schema(reference_relation))
+
+        relations = self.cache.get(normalized_schema)
+        if relations is None:
+            relations = self.list_relations_without_caching(reference_relation)
+            self.cache[normalized_schema] = relations
```

Comment:
Should we update the cache directly here, or override `_get_data_objects` in the `CachingAdapter` and cache it there (but not return from the cache so that `list_relations_without_caching` still works as expected) ?

Reply:
Perhaps we can do it there as well yeah, let me try.

Reply:
On the other hand that will add caching in `list_relations_without_caching`, which contradicts the method name?

Reply:
That's why I suggested to only storing in the cache but not serving from it in `_get_data_objects`:

> (but not return from the cache so that list_relations_without_caching still works as expected)

Reply:
Ah I see, makes sense, let me look what that'd look like 👍

Reply:
Okay, I took a quick look and I'm not really sure about the value of this refactor - it introduces some complexity:

1. need to override `_get_data_objects`,
2. should probably replicate or factor out [logic](https://github.com/TobikoData/sqlmesh/blob/main/sqlmesh/dbt/adapter.py#L229) that exists in `list_relations_without_caching`, so that we can store the appropriate relation objects in the cache.

Also, `_get_data_objects` is used infrequently at the moment, so it looks like we wouldn't get much value back from doing this. The current approach feels a little less involved. Thoughts?